cpp/include/cudf/round_robin.hpp

```diff
@@ -29,8 +29,36 @@ namespace experimental {
  * Returns a new table with rows re-arranged into partition groups and
  * a vector of row offsets to the start of each partition in the output table.
  * Rows are assigned partitions based on their row index in the table,
- * based on the following formula:
- * partition = (row_index + start_partition) % num_partitions
+ * in a round robin fashion.
+ *
+ * @throws cudf::logic_error if num_partitions <= 1 
+ * @throws cudf::logic_error if start_partition >= num_partitions.
```

Comment:
What about negative values of start_partition?

Reply:
What about them? How would you start partitioning from a negative value?

Reply:
Okay, I'm in the process of adding a check `start_partition >= 0`, since `cudf::size_type` is actually an alias to `int32_t`, which _can_ be negative.

Reply:
Done.